packages/types/app/vue.d.ts

```diff
@@ -19,7 +19,7 @@ declare module 'vue/types/options' {
     scrollToTop?: boolean
     transition?: string | Transition | ((to: Route, from: Route) => string)
     validate?(ctx: Context): Promise<boolean> | boolean
-    watchQuery?: boolean | string[]
+    watchQuery?: boolean | string[] | ((to: Route['query'], from: Route['query']) => boolean)
```

Comment:
```suggestion
    watchQuery?: boolean | string[] | ((newQuery: Route['query'], oldQuery: Route['query']) => boolean)
```
Maybe these function param names are more relevant ? dunno